docs/source/usage.md

```diff
@@ -110,5 +110,36 @@ If you have files where tests within the file would interfere with each other, y
 If you want more fine-grained control, you can `export BATS_NO_PARALLELIZE_WITHIN_FILE=true` in `setup_file()`
 or outside any function to disable parallelization only within the containing file.
 
+## Test Coverage
+
+Test coverage can be gathered when using [run][run-test-other-commands] to run
+the code being tested as an external binary; it is not possible to gather test
+coverage for shell functions because shell functions cannot be run by an
+external program, and attempting to do so will likely cause the tests to fail.
+
+Bats does not create or cleanup the test coverage directory, allowing
+accumulation of coverage data over multiple runs.
+
+Only [kcov][kcov] is currently supported:
+
+- To enable test coverage globally use `--kcoverage_dir <directory>`.
```

Comment:
if `kcov` is potentially replaceable with another coverage tool, should the API defined in a more generic way? 
`--coverage-dir` option and `COVERAGE_DIR` arg

Reply:
Sure, I'm happy to change the API - maybe something like this would be better?

- `--coverage_dir` and `$BATS_COVERAGE_DIR`
- `--coverage_tool` and `$BATS_COVERAGE_TOOL`

Then only `gather_coverage` would need changes to support a new coverage tool.